K2Bridge/KustoDAL/KustoResponseParser.cs

```diff
@@ -172,11 +172,26 @@ private void ReportQueryExecutionMetrics(KustoResponseDataSet kustoResponseDataS
                 // Add parent name to aggregations
                 response.AddParentToAgg(parent);
 
+                // Determine how to create the buckets based on the aggregation type
+                Func<DataRow, IBucket> createBucketFromDataRow = null;
+                switch (query.PrimaryAggregation)
```

Comment:
I suppose we cannot do a switch on an object type, it's why you converted it to string ?

Reply:
I confirm, we cannot use typeof in the case statement. But I suppose you could simplify the syntax to get the name of the object type -> obj.GetType().Name.